corehq/apps/reports/view_helpers.py

```diff
@@ -23,15 +23,13 @@ def case_hierarchy_context(case, timezone=None):
         for idx in case.live_indices:
             try:
                 parent_case = idx.referenced_case
-                parent_case.index_info = _index_to_context(idx, is_ancestor=True)
-                parent_cases.append(parent_case)
             except ResourceNotFound:
                 parent_cases.append(None)
-        for parent_case in parent_cases:
-            if parent_case:
-                last_parent_id = parent_case.case_id
-            else:
                 last_parent_id = None
+            else:
+                parent_case.index_info = _index_to_context(idx, is_ancestor=True)
+                parent_cases.append(parent_case)
+                last_parent_id = parent_case.case_id
```

Comment:
nit: I think this could be simplified a bit more
```suggestion
            else:
                parent_case.index_info = _index_to_context(idx, is_ancestor=True)
                parent_cases.append(parent_case)
        last_parent_id = parent_cases[-1].case_id if parent_cases[-1] else None
```

Reply:
Thanks for the feedback - looking at this a little more closely, it seems this is actually only necessary for the main case anyways, as the rest will have it set in `get_case_hierarchy`.  This gets simpler and more self-explanatory if we modify only the main case.

I'm also kinda tempted to set it inside the `else` block here, so it gets set for each existing parent, with the last one winning.  If we even do have missing parents, that seems like the best behavior anyways.  That said, I'll avoid the behavior change for now.  Just pushed a commit.